test/dynamic-import-tests.js

```diff
@@ -0,0 +1,76 @@
+const assert = require("assert");
+
+describe("dynamic import", () => {
+  const canUseToStringTag = typeof Symbol === "function" &&
+    typeof Symbol.toStringTag === "symbol";
+
+  it("should support a standalone import() call", () => {
+    let callCount = 0;
+    const moduleImport = module.import;
+
+    module.import = function (id) {
+      callCount++;
+      return moduleImport.call(this, id);
+    };
+
+    import("./misc/abc");
+    module.import = moduleImport;
+    assert.strictEqual(callCount, 1);
+  });
+
+  it("should resolve as a namespace import", (done) => {
+    import("./misc/abc").then((ns) => {
+      const nsTag = canUseToStringTag ? "[object Module]" : "[object Object]";
+      assert.strictEqual(Object.prototype.toString.call(ns), nsTag);
+      assert.deepEqual(ns, { a: "a", b: "b", c: "c" });
+      done();
+    });
+  });
+
+  it("should support a variable id", (done) => {
+    const id = "./misc/abc";
+    import(id).then((ns) => {
+      assert.deepEqual(ns, { a: "a", b: "b", c: "c" });
+      done();
+    });
+  });
+
+  it("should support a template string id", (done) => {
+    const id = "./misc/abc";
+    import(`${id}`).then((ns) => {
+      assert.deepEqual(ns, { a: "a", b: "b", c: "c" });
+      done();
+    });
+  });
+
+  it("should establish live binding of values", (done) => {
+    import("./misc/live").then((ns) => {
```

Comment:
For future reference, Mocha lets you return a `Promise` from the test function rather than having to call the `done` function.